client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get, noop } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+
+/**
+ * Dispatches a request to save particular onboarding settings on a site
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const saveJetpackOnboardingSettings = ( { dispatch, getState }, action ) => {
+	const { settings, siteId } = action;
+	const state = getState();
+	const token = get( state.jetpackOnboarding.credentials, [ siteId, 'token' ], null );
+	const jpUser = get( state.jetpackOnboarding.credentials, [ siteId, 'userEmail' ], null );
```

Comment:
I think you can drop the `null` defaults since they don't add much value, do they?

Reply:
I agree, but I'm thinking of leaving them be for now - at some point we might want to further validate requests, and we might want to make sure we're passing a `token` and `jpUser`, even though they're `null`. They don't add much value, but they don't add overhead too, right?

Reply:
> at some point we might want to further validate requests, and we might want to make sure we're passing a `token` and `jpUser`, even though they're `null`

Hmm, that doesn't seem to make a whole lot of sense to me, TBH 😄 

This is just me, applying Occam's Razor again; and trying to think of what a reasonable default can be here, functionally speaking. Without the `null` default, the `get`s would just fall back to `undefined`, wouldn't they? Doesn't seem much worse than `null` 😄

Reply:
I disagree here - let me elaborate what I mean. 

Having `token` and `jpUser` as part of the request might help us validate the request is valid. If we have them `null`, they'll be sent as `null` to the endpoint, but they will still be present in the request:

![](https://cldup.com/7WxqNww6rd.png)

If they're `undefined`, they will not be sent to the endpoint:

![](https://cldup.com/7AZja-gF2E.png)

If you want to give it a try, checkout this branch and run the following in your browser console: `dispatch( { type: 'JETPACK_ONBOARDING_SETTINGS_SAVE', siteId: 1234, settings: { siteTitle: 'New Title Here' } } )` (considering that `1234` doesn't correspond to a site that you're currently onboading).

So I'm more inclined to keeping the `null`s here.

Reply:
Hmm, but why would we want to send a `null` `token` or `jpUser` to the endpoint? That's bound to fail, isn't it? So Isn't that something that we should prevent on the client side already?

Reply:
Let me quote myself from above:

> Having `token` and `jpUser` as part of the request might help us validate the request is valid. 

In other words, if `token` and `jpUser` are missing, we can immediately consider the request invalid, even before starting to compare them against valid values. So if we have validation for `token` and `jpUser` existence on the server side, that would just end the validation at a little earlier point than if we were to reach the point where we compare them against the right values.